### PR TITLE
utils: deduce year from volume for JHEP and JCAP

### PR DESCRIPTION
**Pull request by jacquerie**
Closes https://github.com/inspirehep/inspire-schemas/issues/302.

### CODE REVIEW

File: inspire_schemas/utils.py [hunk 1]
```diff
@@ -753,6 +753,8 @@ def convert_old_publication_info_to_new(publication_infos):
         journal_volume = _publication_info.get('journal_volume')
 
         if journal_title in _JOURNALS_WITH_YEAR_ADDED_TO_VOLUME and journal_volume and len(journal_volume) == 4:
+            was_last_century = int(journal_volume[:2]) > 50
```
Comment (jacquerie) [inline inspire_schemas/utils.py]: My apologies to the developer reading this in 2051.

Comment (michamos) [inline inspire_schemas/utils.py]: you could use 90, JHEP and JCAP are from the nineties, but on the other hand, invenio 3 is meant to run 30 years, so 50 is probably enough ;)

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -555,6 +555,34 @@ def test_convert_old_publication_info_to_new_handles_year_added_to_volumes():
...
 
+def test_convert_old_publication_info_to_new_deduces_year_from_volume():
+    schema = utils.load_schema('hep')
+    subschema = schema['properties']['publication_info']
+
+    publication_info = [
+        {
+            'artid': '065',
+            'journal_title': 'JHEP',
+            'journal_volume': '05',
```
Comment (michamos) [inline tests/unit/test_utils.py]: doesn't seem to be testing what's declared.

Comment (jacquerie) [reply tests/unit/test_utils.py]: It does seem that I'm dumb... Fixed.

**Pull request merged**